File: credit-engine-api/src/main/java/com/credit/engine/application/service/pricing/PricingParameterService.java
package com.credit.engine.application.service.pricing;

import com.credit.engine.application.dto.pricing.PricingParameterRequest;
import com.credit.engine.application.dto.pricing.PricingParameterResponse;
import com.credit.engine.domain.model.receivable.ReceivableType;

import java.util.List;

public interface PricingParameterService {

    PricingParameterResponse create(PricingParameterRequest request);

    PricingParameterResponse findCurrent(ReceivableType receivableType);

    List<PricingParameterResponse> findHistory(ReceivableType receivableType);

}

File: credit-engine-ui/src/services/pricing/pricingParameter.service.ts
import { http } from '../../api/client';
import DateUtils from '../../utils/DateUtils';
import type PricingParameterRequest from './dto/PricingParameterRequest';
import type PricingParameterResponse from './dto/PricingParameterResponse';
import type { ReceivableType } from '../receivable/ReceivableType';

const controllerUrl = '/pricing-parameters';

export const pricingParameterService = {

    create(request: PricingParameterRequest) {
        return http.post<PricingParameterResponse>(controllerUrl, {
            ...request,
            effectiveDate: request.effectiveDate ? DateUtils.toLocalDateString(request.effectiveDate) : null,
        });
    },

    findCurrent(receivableType: ReceivableType) {
        return http.get<PricingParameterResponse>(`${controllerUrl}/${receivableType}/current`);
    },

    findHistory(receivableType: ReceivableType) {
        return http.get<PricingParameterResponse[]>(`${controllerUrl}/${receivableType}/history`);
    },

};
